### DISCUSSION TITLE
How to connect switch etc. to rpi4

### DISCUSSION THREAD

**Discussion by krisauseu** [Ysurac/openmptcprouter#3681] [Q&A]
Hello,

I really don't understand this in the documentation for Raspberry Pi:

You don't need ethernet adapters, only a switch is enough. Modems doesn't need to be directly connected to RPI ethernet port.

I can't imagine this. Could someone please give me some more details?
First, do I need a managed switch, or a 08/15-switch?
Until now I thought a switch simply splits up a port to more ports. But then it should not be possible to mix WAN and LAN Ports?! Or can OpenWRT handle this?

Please help me with this part, everything else I've just installed and should work.

Thx & regards 
Kris

**Comment by bobradamus** [comment]
Hey Kris, it’s really simple..but not obvious at the start! You need a cheap unmanged switch. You might be thinking of an old hub, a switch re-broadcasts from one to all in the same subnet (I think thats a reasonable description but not perfect).

Pi4 ethernet connects to the switch, your computer computer (the device you want to use the bonded connection) connects to another port on the same switch.

Then, each of the connections you want to bond together connect to the switch too. You must turn the dchp server off on each of those devices and assign IP addresses that don’t conflict. Each of my routers are 192.168.10.1 / 20.1 / 30.1, and then yu access the OMR router on 192.168.100.1 on your computer. There’s no on-screen config for the RPi4 directly, treat it as headless unless you want to see the kernal logs on screen.

Thanks, Rob

**Reply by krisauseu** [reply]
Thank you for your explanation..

Just another question: one of my connections is Starlink.. on the first look there's no way to change its IP , I'm using this in bypass mode.
![Screenshot_20241202-233456](https://github.com/user-attachments/assets/8c50ce8c-e5c2-4181-8bde-7469b7f5686a)

That's what I see when entering 192.168.100.1 in the browser. Is that starlinks IP? Then I have to change the IP of omr, because it's the same !?

**Reply by bobradamus** [reply]
full disclosure, i’ve not done this with starlink yet. But yes, disconnect starlink and go to 192.168.100.1 and login to OMR, you can change its IP on the main wizard I think..

**Comment by krikey** [comment]
I've just bought a RPi5 and two USB -> Ethernet adapters with the intention of plugging two VDSL routers into each of the USB adapters and the RPI5s built in ethernet will be for my LAN.

Are you saying that you can run OMR with just the built in ethernet port by plugging the routers into the same switch but just using different and non-clashing subnets?

**Reply by Ysurac** [reply]
Yes. No need for adapters if you don't use DHCP in routers.
